CMakeLists.txt

```diff
@@ -571,6 +565,20 @@ if(NOT AVIF_CODEC_AOM_ENABLED
     message(WARNING "libavif: No decoding library is enabled.")
 endif()
 
+if(AVIF_USE_CXX
+   OR (AVIF_BUILD_APPS AND AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
+   OR AVIF_ENABLE_FUZZTEST
+   OR AVIF_ENABLE_GTEST
+   OR (AVIF_CODEC_AOM_ENABLED AND (AOM_INTERFACE_LINK_LIBRARIES MATCHES vmaf))
```

Comment:
IMPORTANT: With the new meaning of `AVIF_USE_CXX` defined at line 136, I would expect that this case falls under the new definition of `AVIF_USE_CXX` and therefore should be handled by setting `AVIF_USE_CXX` to `ON`.

However, the original comment has a comment (which may have been deleted accidentally in this pull request):

```
if(AVIF_CODEC_AOM_ENABLED)
    get_target_property(AOM_INTERFACE_LINK_LIBRARIES aom INTERFACE_LINK_LIBRARIES)
    if(AOM_INTERFACE_LINK_LIBRARIES MATCHES vmaf)
        # vmaf only affects the avifenc and avifdec targets below, not the avif library nor examples targets.
        set(AVIF_USE_CXX ON)
    endif()
endif()
```

If that comment is accurate, then we do NOT need to link the libavif library using the C++ compiler. So we need to look into this. If we are not sure if that comment is wrong, it is best to assume the comment is accurate and preserve the original behavior.

Reply:
Bad copy paste: this will be fixed once https://github.com/AOMediaCodec/libavif/issues/2274 is fixed

Reply:
I looked up the source of that comment. The comment was originally added in https://github.com/AOMediaCodec/libavif/commit/ef6da2a78 and updated in https://github.com/AOMediaCodec/libavif/commit/47e58af14. I believe the comment was added to describe the existing code. The code was originally added in https://github.com/AOMediaCodec/libavif/commit/778b77fa0. I think the code is incorrect. The comment accurately describes the code. That's why I find the comment confusing and seeminly incorrect.

This pull request corrects the code and deletes the comment (which becomes inaccurate because the code it described has changed). That's good.